File: src/main/java/com/example/app/service/NoteService.java
package com.example.app.service;

import com.example.app.utils.model.entities.Note;
import lombok.NonNull;

import java.util.List;

/*
После добавления заметки пользователь получает обратно эту класс с данной заметкой
 и заполненным id, который будет отправлять для удаления и обновления существующей заметки
 */


public interface NoteService {

    Note addNote(@NonNull Note note);

    Note updateNote(@NonNull Note note);

    void deleteNote(@NonNull Long id);

    List<Note> getAllNotes();
}

File: src/main/kotlin/com/example/service/NoteService.kt
package com.example.service

import com.example.utils.dto.entities.Note
import org.springframework.stereotype.Service

/*
После добавления заметки пользователь получает обратно эту класс с данной заметкой
и заполненным id, который будет отправлять для удаления и обновления существующей заметки
*/
@Service
interface NoteService {
    fun addNote(note: Note): Note
    fun updateNote(note: Note): Note
    fun deleteNote(id: Long)

    fun allNotes(): List<Note>
}
